hassio/src/dialogs/snapshot/dialog-hassio-create-snapshot.ts

```diff
@@ -0,0 +1,321 @@
+        ${this._snapshotHasPassword
+          ? html`
+              <paper-input
+                .label=${this._dialogParams.supervisor.localize(
+                  "snapshot.password"
+                )}
+                type="password"
+                name="snapshotPassword"
+                .value=${this._snapshotPassword}
+                @value-changed=${this._handleTextValueChanged}
+              >
+              </paper-input>
+            `
+          : undefined}
+        ${this._error !== ""
+          ? html` <p class="error">${this._error}</p> `
+          : undefined}
+        <mwc-button slot="secondaryAction" @click=${this._closeDialog}>
+          ${this._dialogParams.supervisor.localize("common.close")}
+        </mwc-button>
+        <ha-progress-button slot="primaryAction" @click=${this._createSnapshot}>
+          ${this._dialogParams.supervisor.localize("snapshot.create")}
+        </ha-progress-button>
+      </ha-dialog>
+    `;
+  }
+
+  private _handleTextValueChanged(ev: PolymerChangedEvent<string>) {
+    const input = ev.currentTarget as PaperInputElement;
+    this[`_${input.name}`] = ev.detail.value;
+  }
+
+  private _handleCheckboxValueChanged(ev) {
+    const input = ev.currentTarget as PaperCheckboxElement;
+    this[`_${input.name}`] = input.checked;
+  }
+
+  private _handleRadioValueChanged(ev: PolymerChangedEvent<string>) {
+    const input = ev.currentTarget as PaperRadioGroupElement;
+    this[`_${input.getAttribute("name")}`] = ev.detail.value;
+  }
+
+  private _folderChecked(ev) {
+    const { idx, checked } = ev.currentTarget!;
+    this._folderList = this._folderList.map((folder, curIdx) =>
+      curIdx === idx ? { ...folder, checked } : folder
+    );
+  }
+
+  private _addonChecked(ev) {
+    const { idx, checked } = ev.currentTarget!;
+    this._addonList = this._addonList.map((addon, curIdx) =>
+      curIdx === idx ? { ...addon, checked } : addon
+    );
+  }
+
+  private async _createSnapshot(ev: CustomEvent): Promise<void> {
+    if (this._dialogParams!.supervisor.info.state !== "running") {
+      await showAlertDialog(this, {
+        title: this._dialogParams!.supervisor.localize(
+          "snapshot.could_not_create"
+        ),
+        text: this._dialogParams!.supervisor.localize(
+          "snapshot.create_blocked_not_running",
+          "state",
+          this._dialogParams!.supervisor.info.state
+        ),
+      });
+    }
+    const button = ev.currentTarget as any;
+    button.progress = true;
+
+    this._error = "";
+    if (this._snapshotHasPassword && !this._snapshotPassword.length) {
+      this._error = this._dialogParams!.supervisor.localize(
+        "snapshot.enter_password"
+      );
+      button.progress = false;
+      return;
+    }
+    const name =
+      this._snapshotName ||
+      new Date().toLocaleDateString(navigator.language, {
+        weekday: "long",
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+      });
```

Comment:
Use the `formatDate` helper, it has a fallback as not all browsers support `toLocaleDateString`